test/e2e/flowaggregator_test.go

```diff
@@ -226,71 +242,99 @@ func checkRecordsForFlows(t *testing.T, data *TestData, srcIP string, dstIP stri
-						}
+					if !strings.Contains(record, "antrea-test/perftest-c") {
+						t.Errorf("Record with ServiceIP does not have Service name")
 					}
 				}
-				if checkNetworkPolicy {
-					// Check if records have both ingress and egress network policies.
-					if !strings.Contains(record, ingressNetworkPolicyName) {
-						t.Errorf("Record does not have NetworkPolicy name with ingress rule")
-					}
-					if !strings.Contains(record, fmt.Sprintf("%s: %s", "ingressNetworkPolicyNamespace", testNamespace)) {
-						t.Errorf("Record does not have correct ingressNetworkPolicyNamespace")
-					}
-					if !strings.Contains(record, egressNetworkPolicyName) {
-						t.Errorf("Record does not have NetworkPolicy name with egress rule")
-					}
-					if !strings.Contains(record, fmt.Sprintf("%s: %s", "egressNetworkPolicyNamespace", testNamespace)) {
-						t.Errorf("Record does not have correct egressNetworkPolicyNamespace")
-					}
+			}
+			if checkNetworkPolicy {
+				// Check if records have both ingress and egress network policies.
+				if !strings.Contains(record, ingressNetworkPolicyName) {
+					t.Errorf("Record does not have NetworkPolicy name with ingress rule")
+				}
+				if !strings.Contains(record, fmt.Sprintf("%s: %s", "ingressNetworkPolicyNamespace", testNamespace)) {
+					t.Errorf("Record does not have correct ingressNetworkPolicyNamespace")
+				}
+				if !strings.Contains(record, egressNetworkPolicyName) {
+					t.Errorf("Record does not have NetworkPolicy name with egress rule")
+				}
+				if !strings.Contains(record, fmt.Sprintf("%s: %s", "egressNetworkPolicyNamespace", testNamespace)) {
+					t.Errorf("Record does not have correct egressNetworkPolicyNamespace")
+				}
+			}
+			// Skip the bandwidth check for the iperf control flow records which have 0 delta count.
+			if checkBandwidth && !strings.Contains(record, "octetDeltaCount: 0") {
+				exportTime := int64(getUnit64FieldFromRecord(t, record, "flowEndSeconds"))
+				curOctetTotalCount := getUnit64FieldFromRecord(t, record, "octetTotalCountFromSourceNode")
+				if curOctetTotalCount > octetTotalCount {
+					octetTotalCount = curOctetTotalCount
 				}
-				// Check the bandwidth using octetDeltaCount in data record.
-				if checkBandwidth {
-					checkBandwidthFromRecord(t, record, bandwidth)
+				curOctetDeltaCount := getUnit64FieldFromRecord(t, record, "octetDeltaCountFromSourceNode")
+				// Check the bandwidth using octetDeltaCountFromSourceNode, if this record
+				// is not either the first record or the last in the stream of records.
+				if curOctetDeltaCount != curOctetTotalCount && exportTime < timeStartSec+iperfTimeSec {
+					t.Logf("Check the bandwidth using octetDeltaCountFromSourceNode in data record.")
+					// This middle record should aggregate two records from Flow Exporter
+					checkBandwidthByInterval(t, bandwidthInMbps, curOctetDeltaCount, float64(2*exporterActiveFlowExportTimeout/time.Second), "octetDeltaCountFromSourceNode")
```

Comment:
could we use flow updated time from the last record and current record as the interval? We do not have to hardcode 2 records.

Reply:
Thanks Srikar. Considering the limitation of using `flowEndSeconds` to calculate bandwidth currently (which I will open an issue to track), I think it may be better to change to use flow updated time of the last record and current record as the interval when fixing that issue in the future, do you agree?

Reply:
Sure there may be a change in the field name. Isn't it easy for us to move from one field name to another? Using the flow updated time to determine the interval is more intuitive IMO.
I do not have a strong opinion. If you have all the tests passed, we could fix this in follow-up PR when fixing the issue.

Reply:
Got it, I have triggered 4 times jenkins test for current commit and get all passed. I prefer fixing it in another PR along with issue, considering current commit has been fully tested.